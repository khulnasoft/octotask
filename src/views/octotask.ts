export function octotaskView({
  name,
  description,
  version,
}: {
  name?: string;
  description?: string;
  version?: string;
}): string {
  return `<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>${name || "Your App"} | built with Octotask</title>
    <link rel="icon" href="/octotask/static/octotask-head.png">
    <link rel="stylesheet" href="/octotask/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/octotask/static/robot.svg" alt="Octotask Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to ${name || "your Octotask App"}
${
  version
    ? `            <span class="Label Label--outline v-align-middle ml-2 text-gray-light">v${version}</span>\n`
    : ""
}        </h1>

          <p>${
            description
              ? description
              : 'This bot was built using <a href="https://github.com/octotask/octotask">Octotask</a>, a framework for building GitHub Apps.'
          }</p>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://octotask.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://github.com/octotask/octotask/discussions" class="btn btn-outline">Discuss on GitHub</a>
        </div>
      </div>
    </div>
  </body>
</html>`;
}
