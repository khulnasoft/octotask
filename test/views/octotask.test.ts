import { describe, expect, test } from "vitest";

import { octotaskView } from "../../src/views/octotask.js";

describe("octotaskView", () => {
  test("not providing parameters", () => {
    expect(octotaskView({})).toMatchSnapshot();
  });

  test("providing name", () => {
    expect(
      octotaskView({
        name: "My App",
      }),
    ).toMatchSnapshot();
  });

  test("providing description", () => {
    expect(
      octotaskView({
        description: "My App with Octotask",
      }),
    ).toMatchSnapshot();
  });

  test("providing description", () => {
    expect(
      octotaskView({
        version: "1.0.0",
      }),
    ).toMatchSnapshot();
  });
});
