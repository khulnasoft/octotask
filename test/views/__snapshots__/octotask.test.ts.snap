// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`octotaskView > not providing parameters 1`] = `
"<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Your App | built with Octotask</title>
    <link rel="icon" href="/octotask/static/octotask-head.png">
    <link rel="stylesheet" href="/octotask/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/octotask/static/robot.svg" alt="Octotask Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to your Octotask App
        </h1>

          <p>This bot was built using <a href="https://github.com/octotask/octotask">Octotask</a>, a framework for building GitHub Apps.</p>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://octotask.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://github.com/octotask/octotask/discussions" class="btn btn-outline">Discuss on GitHub</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;

exports[`octotaskView > providing description 1`] = `
"<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Your App | built with Octotask</title>
    <link rel="icon" href="/octotask/static/octotask-head.png">
    <link rel="stylesheet" href="/octotask/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/octotask/static/robot.svg" alt="Octotask Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to your Octotask App
        </h1>

          <p>My App with Octotask</p>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://octotask.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://github.com/octotask/octotask/discussions" class="btn btn-outline">Discuss on GitHub</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;

exports[`octotaskView > providing description 2`] = `
"<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Your App | built with Octotask</title>
    <link rel="icon" href="/octotask/static/octotask-head.png">
    <link rel="stylesheet" href="/octotask/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/octotask/static/robot.svg" alt="Octotask Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to your Octotask App
            <span class="Label Label--outline v-align-middle ml-2 text-gray-light">v1.0.0</span>
        </h1>

          <p>This bot was built using <a href="https://github.com/octotask/octotask">Octotask</a>, a framework for building GitHub Apps.</p>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://octotask.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://github.com/octotask/octotask/discussions" class="btn btn-outline">Discuss on GitHub</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;

exports[`octotaskView > providing name 1`] = `
"<!DOCTYPE html>
<html lang="en" class="height-full" data-color-mode="auto" data-light-theme="light" data-dark-theme="dark">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>My App | built with Octotask</title>
    <link rel="icon" href="/octotask/static/octotask-head.png">
    <link rel="stylesheet" href="/octotask/static/primer.css">
  </head>
  <body class="height-full bg-gray-light">
    <div class="d-flex flex-column flex-justify-center flex-items-center text-center height-full">
      <img src="/octotask/static/robot.svg" alt="Octotask Logo" width="100" class="mb-6">
      <div class="box-shadow rounded-2 border p-6 bg-white">
        <h1>
          Welcome to My App
        </h1>

          <p>This bot was built using <a href="https://github.com/octotask/octotask">Octotask</a>, a framework for building GitHub Apps.</p>
      </div>

      <div class="mt-4">
        <h4 class="alt-h4 text-gray-light">Need help?</h4>
        <div class="d-flex flex-justify-center mt-2">
          <a href="https://octotask.github.io/docs/" class="btn btn-outline mr-2">Documentation</a>
          <a href="https://github.com/octotask/octotask/discussions" class="btn btn-outline">Discuss on GitHub</a>
        </div>
      </div>
    </div>
  </body>
</html>"
`;
